### ISSUE TITLE
Outgoing payment is incorrectly calling ripple rest when path is empty

### ISSUE THREAD

**Issue by lmwattebled** [RoastBeefSandwichCo/gatewayd#1]
I detected a problem when trying to use record deposit with the following configuration:

User lmwattebled is registered and activated on my gatewayd, with external account equals to 4
Trustlines are defined:
- ~lmwattebled  --> cold_wallet    in Euros (rippling Off)
- hot_wallet --> cold_wallet    in Euros   (rippling Off)

Based on :

```
gateway record_deposit  <amount> <currency> <external_account_id> 
```

I tried to record a deposit with this command:

```
gateway record_deposit  3 EUR 4 <external_account_id> 
```

The deposit was well recorded, but was not successfully transmitted to ripple network when outgoing payment was submitted.
On Ripple-rest logs, I saw repetitive  attempts to submit payment from hot_wallet to ~lmwattebled:

```
2015-08-20T09:51:54.497Z - info: POST /v1/accounts/rBgSUyGjBfKtz1VkeWhVpUEbJYQSRM9Rdu/payments?validated=true 500 161.826 ms - 128
2015-08-20T09:51:54.508Z - info: [DB] Saved transaction: unsubmitted: DAA52A671915C03ED047FD83B66DBE89F981E28799CA736B7E99BBAFB6DBBD08: undefined
2015-08-20T09:51:54.521Z - info: [DB] Saved transaction: submitted: DAA52A671915C03ED047FD83B66DBE89F981E28799CA736B7E99BBAFB6DBBD08: undefined
2015-08-20T09:51:54.534Z - info: [DB] Saved transaction: failed: DAA52A671915C03ED047FD83B66DBE89F981E28799CA736B7E99BBAFB6DBBD08: undefined
```

I analysed logs of my local rippled in trace mode and found:

```
2015-Aug-20 09:51:54 STBase:NFO STPathSet: Empty path.
2015-Aug-20 09:51:54 HandlerLog:DBG Ws:: Sending '{"error":"invalidTransaction","error_exception":"empty path","id":4,"request":{"command":"submit","id":4,"tx_blob":"12....."},"status":"error","type":"response"}'
```

So, I made full analysys of payment path :  I reproduced the payment first with rippled client (without any problem), and then tried with ripple-rest commands.
And I was able to reproduce the same problem: It's coming from the fact that path returned from preparePayment is empty. But submitting an empty path is not supported when submitting payment with Ripple-rest.
You will be able to see detailled tests on Ripple forum [here](https://forum.ripple.com/viewtopic.php?f=2&t=15593), because I had to request help to find what was wrong! :)

It appears that: when path returned by "prepare payment" of ripple-rest is empty, this empy path must not be transmitted when submitting payment.

Please feel free to ask more information if needed.

**Comment by whotooktwarden** [comment]
@ninobrooks It seems as if gatewayd will get confused if paths is undefined when it attempts to submit an outgoing payment.  Isn't the paths field optional?  Can gwd be modified to not return a paths field if it is undefined, elseif the field is set, return the paths field?

**Comment by ninobrooks** [comment]
@whotooktwarden Yes, that's probably the fix.
@lmwattebled Looks like you really nailed it down! I'm not sure I'll ever get a more useful bug report LOL

Would you please confirm for me that the problem exists in this fork? I have tagged a new release for users who are coming from RL's gatewayd to this one. [3.34.2.1](https://github.com/RoastBeefSandwichCo/gatewayd/tags)

**Comment by lmwattebled** [comment]
Indeed, I don't remember how, but it appears that I found your gatewayd sources on githug around three weeks ago, and I took them because this was the most recent ones I found! :)
I downloaded sources from master branch of this github repository on July, 30 th. I don't know how to check the tag associated with it, but release notes in doc subdirectory are mentionning: 3.35.2 (03/12/2015).
Hope this will help you.

**Comment by ninobrooks** [comment]
Try [code]git checkout a3cc109[/code]
If you are on the RBSC fork, it will succeed and you'll get the recently re-added documentation and I'll tag this issue as a bug and get it fixed.
If it fails, you're on another fork.

**Comment by lmwattebled** [comment]
Indeed, I was on a downloaded zip file.
So, I've built a whole new setup based on git this time, and done `git checkout a3cc109`
I'm on this fork now, and reproduced the issue.
Thanks.

**Comment by ninobrooks** [comment]
Tagged. It'll be another day before I can work on this. 

@lmwattebled Thank you for being so helpful with all the information and by changing over to our release to confirm the bug. We'll get it squashed asap.
